internal/token_test.go

```diff
@@ -0,0 +1,63 @@
+package astro
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+)
+
+func TestTokenizer(t *testing.T) {
+	tests := []struct {
+		name string
+		input string
+		want []TokenType
+	}{
+		{
+		  "doctype",
+			`<!DOCTYPE html>`,
+			[] TokenType { DoctypeToken },
```

Comment:
I was thinking to expand these tests, we could test longer HTML strings that yield slices of tokens. Hence the param here being a slice.

What do you think about the test setup here?